xbmc/utils/StringUtils.cpp

```diff
@@ -1071,6 +1075,51 @@ int64_t StringUtils::AlphaNumericCompare(const wchar_t* left, const wchar_t* rig
   wchar_t lc, rc;
   int64_t lnum, rnum;
   bool lsym, rsym;
+
+#ifdef TARGET_ANDROID
+  if (g_langInfo.UseLocaleCollation())
+  {
+    UErrorCode ustatus = U_ZERO_ERROR;
+    try
+    {
+      auto ucoll = std::unique_ptr<UCollator, decltype(&ucol_close)>(
+          ucol_open(g_langInfo.GetISOLocale().c_str(), &ustatus), &ucol_close);
+      if (U_FAILURE(ustatus))
+        throw std::runtime_error("StringUtils: ucol_open failed.");
+      ucol_setAttribute(ucoll.get(), UCOL_NUMERIC_COLLATION, UCOL_ON, &ustatus);
+      if (U_FAILURE(ustatus))
+        throw std::runtime_error("StringUtils: failed to set UCOL_NUMERIC_COLLATION.");
+
+      std::vector<UChar> uleft(2 * wcslen(left) + 1);
+      std::vector<UChar> uright(2 * wcslen(right) + 1);
+
+      int32_t ulsize{0};
+      int32_t ursize{0};
+
+      u_strFromWCS(&uleft.front(), uleft.size(), &ulsize, left, -1, &ustatus);
+      if (U_FAILURE(ustatus))
+        throw std::runtime_error("StringUtils: failed to convert left string to UTF16.");
+      u_strFromWCS(&uright.front(), uright.size(), &ursize, right, -1, &ustatus);
+      if (U_FAILURE(ustatus))
+        throw std::runtime_error("StringUtils: failed to convert right string to UTF16.");
```

Comment:
seems a bit unnecessary to use exceptions here. just put the android stuff into a separate function from which you can return early.